src/cpp/core/system/PosixSystem.cpp

```diff
@@ -1680,7 +1744,76 @@ Error launchChildProcess(std::string path,
    return Success() ;
 }
 
+bool filterNonChildProcesses(pid_t pid, pid_t ppid, pid_t pgrp,
+                             const rstudio::core::system::ProcessInfo& process)
```

Comment:
Nit: we usually just use `core::system` rather than `rstudio::core::system` for brevity (the rstudio namespace is already in scope here)